updater/lib/dependabot/pull_request.rb

```diff
@@ -61,19 +56,25 @@ def removed?
     sig { returns(T::Array[Dependency]) }
     attr_reader :dependencies
 
+    sig { returns(T.nilable(Integer)) }
+    attr_reader :pr_number
+
     sig { params(attributes: T::Hash[Symbol, T.untyped]).returns(T::Array[Dependabot::PullRequest]) }
     def self.create_from_job_definition(attributes)
       attributes.fetch(:existing_pull_requests).map do |pr|
+        # Extract pr_number from PR level, or fallback to first dependency if available
+        pr_number = pr.first&.fetch("pr-number", nil)&.to_i
```

Comment:
To handle a new data format and the old one, something like this might work. 

```ruby
pr_number = case pr
  when Array
    pr.first["pr-number"]
  when Hash
    pr["pr-number"]
  end
```